### PR TITLE
Integrate SpeziLLM C++ functionality

### PR DESCRIPTION
**Pull request by philippzagar**
# Integrate SpeziLLM C++ functionality

## :recycle: Current situation & Problem
Currently, the SpeziLLM package contains certain C++ bridging headers in order to utilize functionality from llama.cpp in Swift. However, this requires setting the language standard (C++11) within the SPM Package.swift file as an unsafeFlag, which inherently leads to the package not being able to be versioned.

## :gear: Release Notes 
- Integrate SpeziLLM C++ bridging functionality into StanfordBDHG llama.cpp fork
- Sync with [upstream llama.cpp b1592](https://github.com/ggerganov/llama.cpp/releases/tag/b1592)

## :books: Documentation
--

## :white_check_mark: Testing
Locally tested

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordBDHG/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordBDHG/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: examples/llama.swiftui/llama.cpp.swift/LibLlama.swift
Comment (PSchmiedmayer) [inline examples/llama.swiftui/llama.cpp.swift/LibLlama.swift:1]: @philippzagar Is there anything for us to learn from the additional example they pushed in the repo?

File: swift/include/tokenize.h
Comment (PSchmiedmayer) [inline swift/include/tokenize.h:1]: 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Looks good!

File: examples/llama.swiftui/llama.cpp.swift/LibLlama.swift
Comment (philippzagar) [reply examples/llama.swiftui/llama.cpp.swift/LibLlama.swift:1]: @PSchmiedmayer Thanks for pointing that out, actually missed that!
Sadly, this is just a very very basic integration of llama.cpp without any optimizations or configurations, our implementation is definitely more advanced.

Comment (philippzagar) [reply examples/llama.swiftui/llama.cpp.swift/LibLlama.swift:1]: The committed code builds upon a repo I discovered quite a while ago, see here: https://github.com/bachittle/llama.cpp/tree/swiftui_metal

**Pull request merged**